asst1/asst1.py

```diff
@@ -0,0 +1,64 @@
+import collections
+import string
+
+def swapchars (w) :
+  temp_str = w.translate(None, string.punctuation)
+  temp_str = temp_str.translate(None, " ")
+  freq = collections.Counter(temp_str.lower()).most_common()
+  most, num = freq[0]
+  least, num1 = freq[len(freq) - 1]
+  print most
+  print least
+  ret = ""
+  for c in w :
+    if c == most.upper() :
+      ret += least.upper()
+    elif c == most.lower() :
+      ret += least.lower()
+    elif c == least.upper() :
+      ret += most.upper()
+    elif c == least.lower() :
+      ret += most.lower()
+    else :
+      ret += c
+  return ret
+
+def sortcat (n, *args) :
+  xs = []
+  str = ""
+  i = n
+  for a in args :
+    xs.append(a)
+  xs = sorted(xs, key=len)
+  for a in xs :
+    if i > 0 :
+      str += xs[i]
+      i-=1
+  if n == -1 :
+    for a in reversed(xs) :
+      str += a
+  return str
+
+def bluesclues (abbr) :
+  f = open('states.txt')
+  answer = {}
+  for line in f :
+    k, v = line.strip().split(',')
+    answer[v.strip()] = k.strip()
+  return answer[abbr]
+
+def bluesbooze (name) :
+  f = open('states.txt')
```

Comment:
how could you have used 'bluesclues' to implement 'bluesbooze?' there's a lot of repeat code here that you probably could have avoided